component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java

```diff
@@ -683,4 +700,64 @@ private boolean validateJTIInCache(String jti, SignedJWT signedJWT, JWTCacheEntr
         }
         return true;
     }
+
+    /**
+     * Validate whether the request signing algorithm is configured for the application.
+     *
+     * @param signedJWT     The signed JWT.
+     * @param clientId      Client ID of the application.
+     * @return whether the request signing algorithm is configured for the application.
+     * @throws OAuthClientAuthnException OAuth Client Authentication Exception.
+     */
+    private boolean isValidSignatureAlgorithm(SignedJWT signedJWT, String clientId) throws OAuthClientAuthnException {
+
+        //   Obtain the signing algorithm used to sign the JWT in the request.
+        String requestSigningAlgorithm = signedJWT.getHeader().getAlgorithm().getName();
+        //   Obtain the signing algorithm configured for the application.
+        List<String> configuredSigningAlgorithms = getConfiguredSigningAlgorithm(clientId);
+        //   Mandating FAPI specified JWT signing algorithms.
+        List<String> fapiAllowedSigningAlgorithms = IdentityUtil.getPropertyAsList(FAPI_SIGNATURE_ALG_CONFIGURATION);
+        if (!fapiAllowedSigningAlgorithms.contains(requestSigningAlgorithm)) {
+            if (log.isDebugEnabled()) {
+                log.debug("FAPI unsupported signing algorithm " + requestSigningAlgorithm +
+                        " is used to sign the JWT.");
+            }
+            return false;
+        }
```

Comment:
With https://github.com/wso2-extensions/identity-oauth-addons/pull/108/files#r1348854411 lets have this condition inside the isFapiConformantApp check

Reply:
Fixed in f62c438a62cda25d10b408fb266521c3cd30aaec and 393f070cb3d7c0feb53e15516f6e562bddb36dee